targets/TARGET_RDA/TARGET_UNO_91H/serial_api.c

```diff
@@ -0,0 +1,425 @@
+#include "mbed_assert.h"
+#include <math.h>
+#include <string.h>
+#include <stdlib.h>
+
+#include "serial_api.h"
+#include "cmsis.h"
+#include "pinmap.h"
+#include "gpio_api.h"
+#if DEVICE_SLEEP
+#include "rda_sleep_api.h"
+#endif
+
+/******************************************************************************
+ * INITIALIZATION
+ ******************************************************************************/
+#define UART_NUM    2
+
+#define UART_CLKGATE_REG        (RDA_SCU->CLKGATE0)
+
+#define UART1_CLKEN_MASK        (0x01UL << 21)
+#define RXFIFO_EMPTY_MASK       (0x01UL << 0)
+#define TXFIFO_FULL_MASK        (0x01UL << 19)
+#define AFCE_MASK               (0x01UL << 5)
+
+static const PinMap PinMap_UART_TX[] = {
+    {PA_1,  UART_0, 0},
+    {PB_2,  UART_1, 5},
+    {PD_3,  UART_1, 2},
+    {NC   , NC    , 0}
+};
+
+static const PinMap PinMap_UART_RX[] = {
+    {PA_0,  UART_0, 0},
+    {PB_1,  UART_1, 5},
+    {PD_2,  UART_1, 2},
+    {NC   , NC    , 0}
+};
+
+static const PinMap PinMap_UART_RTS[] = {
+    {PD_1,  UART_1, 2},
+    {NC,    NC,     0}
+};
+
+static const PinMap PinMap_UART_CTS[] = {
+    {PD_0,  UART_1, 2},
+    {NC,    NC,     0}
+};
+
+static uart_irq_handler irq_handler[UART_NUM];
+#if DEVICE_SLEEP
+static sys_wakeup_handler uart_wakeup_handler = NULL;
+#endif
+
+int stdio_uart_inited = 0;
+serial_t stdio_uart;
+
+struct serial_global_data_s {
+    uint32_t serial_irq_id;
+    gpio_t sw_rts, sw_cts;
+    uint8_t count, rx_irq_set_flow, rx_irq_set_api;
+};
+
+static struct serial_global_data_s uart_data[UART_NUM];
+
+void serial_init(serial_t *obj, PinName tx, PinName rx)
+{
+    int is_stdio_uart = 0;
+
+    // determine the UART to use
+    UARTName uart_tx = (UARTName)pinmap_peripheral(tx, PinMap_UART_TX);
+    UARTName uart_rx = (UARTName)pinmap_peripheral(rx, PinMap_UART_RX);
+    UARTName uart = (UARTName)pinmap_merge(uart_tx, uart_rx);
+    MBED_ASSERT((int)uart != NC);
+
+    switch (uart) {
+        case UART_0:
+            obj->index = 0;
+            break;
+        case UART_1:
+            obj->index = 1;
+            /* Enable clock-gating */
+            UART_CLKGATE_REG |= UART1_CLKEN_MASK;
+            break;
+    }
+
+    obj->uart = (RDA_UART_TypeDef *)uart;
+
+    // enable fifos and default rx trigger level
+    obj->uart->FCR = 0 << 0  //FIFO Enable - 0 = Disables, 1 = Enabled
+                   | 0 << 1  // Rx Fifo Reset
+                   | 0 << 2  // Tx Fifo Reset
+                   | 0 << 6; // Rx irq trigger level - 0 = 1 char, 1 = 4 chars, 2 = 8 chars, 3 = 14 chars
+
+    // disable irqs
+    obj->uart->IER = 0 << 0  // Rx Data available irq enable
+                   | 0 << 1  // Tx Fifo empty irq enable
+                   | 0 << 2; // Rx Line Status irq enable
+
+    obj->uart->MCR =  1 << 8; //select clock
+    obj->uart->FRR = 0x2001; //tx_trigger = 0x10, rx_trigger = 0x01
+
+    // set default baud rate and format
+    //serial_baud  (obj, 921600);
```

Comment:
this can be removed? the comment as well about the baud rate.

Reply:
removed it.